cli/src/migrate.js

```diff
@@ -0,0 +1,354 @@
+    rdb_user: parsed.rdb_user || confOptions.rdb_user || 'admin',
+    rdb_password: parsed.rdb_password || confOptions.rdb_password || '',
+    skip_backup: parsed.skip_backup || false,
+    start_rethinkdb: parsed.start_rethinkdb || false,
+  };
+  // sets rdb_host and rdb_port from connect if necessary
+  if (parsed.connect) {
+    config.parse_connect(parsed.connect, options);
+  }
+
+  if (options.project_name == null) {
+    throw new Error('project_name is null');
+  }
+  return options;
+}
+
+function setup() {
+  // Start rethinkdb server if necessary
+  // Connect to whatever rethinkdb server we're using
+  white('Setup');
+  return Promise.resolve().then(() => {
+    if (this.options.project_path && this.options.project_path !== '.') {
+      green(` ├── Changing to directory ${this.options.project_path}`);
+      change_to_project_dir(this.options.project_path);
+    }
+  }).then(() => {
+    // start rethinkdb server if necessary
+    if (this.options.start_rethinkdb) {
+      green(' ├── Starting RethinkDB server');
+      return start_rdb_server({ quiet: true }).then((server) => {
+        this.rdb_server = server;
+        this.options.rdb_host = 'localhost';
+        this.options.rdb_port = server.driver_port;
+      }).then(() => Promise.delay(2000));
+    } else {
+      return undefined;
+    }
+  }).then(() => {
+    green(' ├── Connecting to RethinkDB');
+    return r.connect({
+      host: this.options.rdb_host,
+      port: this.options.rdb_port,
+      user: this.options.rdb_user,
+      password: this.options.rdb_password,
+    });
+  }).then((conn) => {
+    green(' └── Successfully connected');
+    this.conn = conn;
+  });
+}
+
+function teardown() {
+  return Promise.resolve().then(() => {
+    white('Cleaning up...');
+    // close the rethinkdb connection
+    if (this.conn) {
+      green(' ├── Closing rethinkdb connection');
+      return this.conn.close();
+    } else {
+      return undefined;
+    }
+  }).then(() => {
+    // shut down the rethinkdb server if we started it
+    if (this.rdb_server) {
+      green(' └── Shutting down rethinkdb server');
+      return this.rdb_server.close();
+    } else {
+      return undefined;
+    }
+  });
+}
+
+function validateMigration() {
+  // check that `${project}_internal` exists
+  const project = this.options.project_name;
+
+  const checkForHzTables = r.db('rethinkdb')
+          .table('table_config')
+          .filter({ db: project })('name')
+          .contains((x) => x.match('^hz_'))
+          .branch(r.error(`Some tables in ${project} have an hz_ prefix`), true)
+
+  return Promise.resolve().then(() => {
+    white('Validating current schema version');
+    return r.dbList().contains(`${project}_internal`)
+      .branch(true, r.error(`${project}_internal not found`))
+      .do(() => checkForHzTables)
+      .run(this.conn)
+      .then(() => green(' └── Pre-2.0 schema found'))
+      .catch(() => {
```

Comment:
This could fail for a number of reasons, should probably wrap the error rather than discard it.